index.js

```diff
@@ -1,28 +1,52 @@
 'use strict';
-// vim: set expandtab:
-var request = require('request').defaults({ json: true });
 var url = require('url');
+var hyperquest = require('hyperquest');
+var bl = require('bl');
 var nub = require('nub');
 
+function tryit(fn) {
+  try {
+    return fn();
+  } catch (err) {
+    return err;
+  }
+}
+
 function getPkgs(registry, cb) {
-  request(url.resolve(registry + '/', '-/_view/allVersions?reduce=false'), function(err, res, body) {
+  var req = hyperquest({
+    uri: url.resolve(registry + '/', '-/_view/allVersions?reduce=false')
+  });
+  req.on('error', cb);
+
+  req.pipe(bl(function (err, data) {
     if (err) {
-      return cb(err);
+      cb(err);
+      return;
     }
 
-    if (res.statusCode !== 200) {
-      err = new Error('received non-200 status code');
-      return cb(err);
+    if (req.response.statusCode !== 200) {
+      cb(new Error('received non-200 status code'));
+      return;
+    }
+
+    var body = tryit(function () {
```

Comment:
[`json-try-parse`](https://www.npmjs.com/package/json-try-parse) might be useful. I'd be willing to alter it to return the error since later versions of node started to produce useful errors here.